### ISSUE TITLE
cfn-lint failing with E3002 for VpcEndpointType

### ISSUE THREAD

**Issue by kosala-bnz** [awslabs/cfn-python-lint#500]
*cfn-lint version: (`cfn-lint --version`)*
0.8.2

*Description of issue.*
When defining VPC interface endpoint  CFN lint fails with E3002 for VpcEndpointType = "Interface" 

Please provide as much information as possible:
* Template linting issues:
[
    {
        "Rule": {
            "Id": "E3002",
            "Description": "Making sure that resources properties are properly configured",
            "ShortDescription": "Resource properties are valid",
            "Source": "https://github.com/awslabs/cfn-python-lint/blob/master/docs/cfn-resource-specification.md#properties"
        },
        "Location": {
            "Start": {
                "ColumnNumber": 7,
                "LineNumber": 227
            },
            "End": {
                "ColumnNumber": 22,
                "LineNumber": 227
            }
        },
        "Level": "Error",
        "Message": "Invalid Property Resources/rKMSendpointA/Properties/VpcEndpointType",
        "Filename": "c:/KMS/kms-endpoint.yaml"
    },
    {
        "Rule": {
            "Id": "E3002",
            "Description": "Making sure that resources properties are properly configured",
            "ShortDescription": "Resource properties are valid",
            "Source": "https://github.com/awslabs/cfn-python-lint/blob/master/docs/cfn-resource-specification.md#properties"
        },
        "Location": {
            "Start": {
                "ColumnNumber": 7,
                "LineNumber": 248
            },
            "End": {
                "ColumnNumber": 22,
                "LineNumber": 248
            }
        },
        "Level": "Error",
        "Message": "Invalid Property Resources/rKMSendpointB/Properties/VpcEndpointType",
        "Filename": "c:/KMS/kms-endpoint.yaml"
    },
    {
        "Rule": {
            "Id": "E3002",
            "Description": "Making sure that resources properties are properly configured",
            "ShortDescription": "Resource properties are valid",
            "Source": "https://github.com/awslabs/cfn-python-lint/blob/master/docs/cfn-resource-specification.md#properties"
        },
        "Location": {
            "Start": {
                "ColumnNumber": 7,
                "LineNumber": 268
            },
            "End": {
                "ColumnNumber": 22,
                "LineNumber": 268
            }
        },
        "Level": "Error",
        "Message": "Invalid Property Resources/rKMSendpointC/Properties/VpcEndpointType",
        "Filename": "c:/KMS/kms-endpoint.yaml"
    }
]

  * Please provide a CloudFormation sample that generated the issue.
---
AWSTemplateFormatVersion: 2010-09-09
Description: 'Creating KMS endpoint in the downstream account'
# Created date: 30-11-2018 
# Author: Kosala Abeysekera 
# Version: 0.1
# JIRA: https://digital-dev.nz.thenational.com/jira/browse/ECSAWS-463
# Run at: all dependent (downstream) accounts
# Pre Requisities: 
# This template based on: 

Metadata:
#  cfn-lint:

#    config:

#      ignore_checks:

 #     - E3002
  Stack:
    Value: 1
  VersionDate:
    Value: 20181130
  Identifier:
    Value: template-endpoint-service-creation
  Input:
    Description: VPCID, Route-tableID
  Output:
    Description: Endpoint-ID
  AWS::CloudFormation::Interface:
    ParameterGroups:
    - Label:
        default: KMS-Endpoint 

    - Label:
        default: Network (VPCID and Route)
      Parameters:
      
      - pProductionVPC
   
# AZ1 Network Configurations 
      - pSubnetAAZ1
      - pSubnetBAZ1
      - pSubnetCAZ1

Parameters:

  pProductionVPC:
    Description: Production VPC
    Type: AWS::EC2::VPC::Id

  pSubnetAAZ1:
    Description: Subnet ID's for the Subnet A in Availibility Zones 
    Type: List<AWS::EC2::Subnet::Id>
  pSubnetBAZ1:
    Description: Subnet ID's for the Subnet B in Availibility Zones - Leave false if there is no Subnets 
    Type:  CommaDelimitedList
    Default: "False , False , False"

  pSubnetCAZ1:
    Description: Subnet ID for the Subnet C in Availibility Zone 1 - Leave false if there is no Subnets 
 
    Type:  CommaDelimitedList
    Default: "False , False , False"

Conditions:
# Create 2nd KMS endpoint 
   cKMSEPB: !Not [!Equals [ "False" , !Select [ "0" , !Ref pSubnetBAZ1]]]
# Create 3rd KMS endpoint 
   cKMSEPC: !Not [!Equals [ "False" , !Select [ "0" , !Ref pSubnetCAZ1]]]
   

Resources:
 

  rSecurityGroupKMSEP:
    Type: AWS::EC2::SecurityGroup
    Properties:
      GroupDescription: Security group for KMS-Endpoint
      VpcId: !Ref pProductionVPC
      SecurityGroupIngress:
      - IpProtocol: tcp
        FromPort: 443
        ToPort: 443
        CidrIp: 13.210.0.0/15
      - IpProtocol: tcp
        FromPort: 443
        ToPort: 443
        CidrIp: 13.54.0.0/15
      - IpProtocol: tcp
        FromPort: 443
        ToPort: 443
        CidrIp: 52.95.241.0/24
      - IpProtocol: tcp
        FromPort: 443
        ToPort: 443
        CidrIp: 52.65.0.0/16
      - IpProtocol: tcp
        FromPort: 443
        ToPort: 443
        CidrIp: 54.206.0.0/16
      - IpProtocol: tcp
        FromPort: 443
        ToPort: 443
        CidrIp: 52.64.0.0/17
      - IpProtocol: tcp
        FromPort: 443
        ToPort: 443
        CidrIp: 54.153.128.0/17
      - IpProtocol: tcp
        FromPort: 443
        ToPort: 443
        CidrIp: 52.62.0.0/15
      - IpProtocol: tcp
        FromPort: 443
        ToPort: 443
        CidrIp: 54.253.0.0/16
      - IpProtocol: tcp
        FromPort: 443
        ToPort: 443
        CidrIp: 52.94.248.64/28
      - IpProtocol: tcp
        FromPort: 443
        ToPort: 443
        CidrIp: 54.79.0.0/16
      - IpProtocol: tcp
        FromPort: 443
        ToPort: 443
        CidrIp: 52.95.255.16/28
      - IpProtocol: tcp
        FromPort: 443
        ToPort: 443
        CidrIp: 52.64.128.0/17
      - IpProtocol: tcp
        FromPort: 443
        ToPort: 443
        CidrIp: 13.236.0.0/14
      - IpProtocol: tcp
        FromPort: 443
        ToPort: 443
        CidrIp: 54.252.0.0/16
      - IpProtocol: tcp
        FromPort: 443
        ToPort: 443
        CidrIp: 54.66.0.0/16            
      SecurityGroupEgress:
      - IpProtocol: tcp
        FromPort: 443
        ToPort: 443
        CidrIp: 13.210.0.0/15
      - IpProtocol: tcp
        FromPort: 443
        ToPort: 443
        CidrIp: 13.54.0.0/15
      - IpProtocol: tcp
        FromPort: 443
        ToPort: 443
        CidrIp: 52.95.241.0/24
      - IpProtocol: tcp
        FromPort: 443
        ToPort: 443
        CidrIp: 52.65.0.0/16
      - IpProtocol: tcp
        FromPort: 443
        ToPort: 443
        CidrIp: 54.206.0.0/16
      - IpProtocol: tcp
        FromPort: 443
        ToPort: 443
        CidrIp: 52.64.0.0/17
      - IpProtocol: tcp
        FromPort: 443
        ToPort: 443
        CidrIp: 54.153.128.0/17
      - IpProtocol: tcp
        FromPort: 443
        ToPort: 443
        CidrIp: 52.62.0.0/15
      - IpProtocol: tcp
        FromPort: 443
        ToPort: 443
        CidrIp: 54.253.0.0/16
      - IpProtocol: tcp
        FromPort: 443
        ToPort: 443
        CidrIp: 52.94.248.64/28
      - IpProtocol: tcp
        FromPort: 443
        ToPort: 443
        CidrIp: 54.79.0.0/16
      - IpProtocol: tcp
        FromPort: 443
        ToPort: 443
        CidrIp: 52.95.255.16/28
      - IpProtocol: tcp
        FromPort: 443
        ToPort: 443
        CidrIp: 52.64.128.0/17
      - IpProtocol: tcp
        FromPort: 443
        ToPort: 443
        CidrIp: 13.236.0.0/14
      - IpProtocol: tcp
        FromPort: 443
        ToPort: 443
        CidrIp: 54.252.0.0/16
      - IpProtocol: tcp
        FromPort: 443
        ToPort: 443
        CidrIp: 54.66.0.0/16

      Tags:
      - Key: Name
        Value: bnz-aws-vpcep-apse02-KMS
      - Key: Environment
        Value: Endpoint

  rKMSendpointA:
   Type: AWS::EC2::VPCEndpoint
   Properties: 
      VpcId: !Ref pProductionVPC
      ServiceName: !Join        
      - ''
      - - com.amazonaws.
        - !Ref 'AWS::Region'
        - .kms    

      VpcEndpointType: "Interface"
      SubnetIds: 
        - !Select [ "0", !Ref pSubnetAAZ1]
        - !Select ["1", !Ref pSubnetAAZ1] 
        - !Select ["2", !Ref pSubnetAAZ1]
    
   
      SecurityGroupIds: 
        - !Ref rSecurityGroupKMSEP
 
  rKMSendpointB:
   Type: AWS::EC2::VPCEndpoint
   Condition: cKMSEPB
   Properties: 
      VpcId: !Ref pProductionVPC
      ServiceName: !Join        
      - ''
      - - com.amazonaws.
        - !Ref 'AWS::Region'
        - .kms    
   
      VpcEndpointType: "Interface"
      SubnetIds: 
        - !Select [ "0", !Ref pSubnetBAZ1]
        - !Select ["1", !Ref pSubnetBAZ1] 
        - !Select ["2", !Ref pSubnetBAZ1]
   
      SecurityGroupIds: 
        - !Ref rSecurityGroupKMSEP
 
  rKMSendpointC:
   Type: AWS::EC2::VPCEndpoint
   Condition: cKMSEPC
   Properties: 
      VpcId: !Ref pProductionVPC
      ServiceName: !Join        
      - ''
      - - com.amazonaws.
        - !Ref 'AWS::Region'
        - .kms    
    
      VpcEndpointType: "Interface"
      SubnetIds: 
        - !Select [ "0", !Ref pSubnetCAZ1]
        - !Select ["1", !Ref pSubnetCAZ1] 
        - !Select ["2", !Ref pSubnetCAZ1]
   
      SecurityGroupIds: 
        - !Ref rSecurityGroupKMSEP

Outputs:
  rSecurityGroupKMSEP:
    Value: !Ref rSecurityGroupKMSEP
  rKMSendpointA:
    Value: !Ref rKMSendpointA
  rKMSendpointB:
    Condition: cKMSEPB
    Value: !Ref rKMSendpointB
  rKMSendpointC:
    Condition: cKMSEPC
    Value: !Ref rKMSendpointC
 
...

 

  * If present, please add links to the (official) documentation for clarification.
Type: AWS::EC2::VPCEndpoint
Properties: 
  VpcId: String
  RouteTableIds: 
    - String
  ServiceName: String
  PolicyDocument: String
  VpcEndpointType: String
  PrivateDnsEnabled: Boolean
  SubnetIds: 
    - String
  SecurityGroupIds: 
    - String

https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-ec2-vpcendpoint.html
* Feature request:
  * Please provide argumentation about the missing feature. Context is key!

**Comment by kddejong** [comment]
I tried to replicate this but was unsuccessful.  This template is not producing the same error.  VpcEndpointType did exist in v0.8.2 even though we are v0.9.2 now.  There was something off with that particular attribute in earlier versions.  The cloud specs had it as `VPCEndpointType` up till our v0.8.0 release.  CloudFormation Spec version 2.8.0 (which was bundled in the v0.8.0 release).  You may wish to try `cfn-lint --update-specs` as well to force an update of the specs.  After that can we verify that you are using v0.8.0 or higher?

```yaml
---
AWSTemplateFormatVersion: "2010-09-09"
Resources:
  rKMSendpointC:
    Type: AWS::EC2::VPCEndpoint
    Condition: cKMSEPC
    Properties:
      VpcId: !Ref pProductionVPC
      ServiceName:
        Fn::Join:
        - ''
        - - com.amazonaws.
          - !Ref 'AWS::Region'
          - .kms
      VpcEndpointType: "Interface"
      SubnetIds:
        - !Select [ "0", !Ref pSubnetCAZ1]
        - !Select ["1", !Ref pSubnetCAZ1]
        - !Select ["2", !Ref pSubnetCAZ1]
      SecurityGroupIds:
        - !Ref rSecurityGroupKMSEP
```

**Comment by kosala-bnz** [comment]
Hi Team,

Upgrading the version resolved my issue

Regards
Kosala

**Comment by cmmeyer** [comment]
Great to hear! I'll go ahead and close this issue.
